### PR TITLE
storage api: fix permissions

### PR DESCRIPTION
**Pull request by brauner**
Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by stgraber** [comment]
Good first step, I think what I'd still like to see is;
 - Patch to correct any existing bad permissions
 - Change the directory creation code we have for containers, images and custom volumes so that it's only a Mkdir() call, not a MkdirAll() call as I'd rather have LXD fail to create a container when storage pool creation went bad than end up with some wrong permissions inside storage-pools/*. That'd also likely surface any such bug pretty quickly and in a more debuggable way.

**Comment by brauner** [comment]
> Good first step, I think what I'd still like to see is;

That's why it was labeled as `[WIP [TESTING]]`.

**Comment by brauner** [comment]
@stgraber, good to review. But only if this is urgent. ;)

File: lxd/patches.go [hunk 1]
```diff
@@ -2695,6 +2696,143 @@ func patchDevicesNewNamingScheme(name string, d *Daemon) error {
...
+
+			cuMntPoint := getStoragePoolVolumeMountPoint(poolName, vol)
+			err = os.Chmod(cuMntPoint, 0711)
+			if err != nil && !os.IsNotExist(err) {
+				return err
+			}
+		}
+	}
+
+	cRegular, err := d.cluster.ContainersList(db.CTypeRegular)
```
Comment (stgraber) [inline lxd/patches.go]: @freeekanayaka that will get us all the containers not just the local ones, right?
What's the equivalent to get the local containers as that's what we should be using here.

Comment (freeekanayaka) [reply lxd/patches.go]: Correct, this gets us all containers, not just the local ones.

I don't think there's an equivalent for local nodes only, but it should look something like this:

```go
// ContainersNodeList returns the names of all the containers of the given type on the local node.
func (c *Cluster) ContainersNodeList(cType ContainerType) ([]string, error) {
	q := fmt.Sprintf("SELECT name FROM containers WHERE type=? AND node_id=? ORDER BY name")
	inargs := []interface{}{cType,c.nodeID}
	var container string
	outfmt := []interface{}{container}
	result, err := queryScan(c.db, q, inargs, outfmt)
	if err != nil {
		return nil, err
	}

	var ret []string
	for _, container := range result {
		ret = append(ret, container[0].(string))
	}

	return ret, nil
}
```

Comment (brauner) [reply lxd/patches.go]: @freeekanayaka, can you please send a patch for this?

**Pull request merged**